core/src/main/java/io/questdb/cairo/PartitionPurgeJob.java

```diff
@@ -0,0 +1,210 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.cairo;
+
+import io.questdb.MessageBus;
+import io.questdb.log.Log;
+import io.questdb.log.LogFactory;
+import io.questdb.mp.RingQueue;
+import io.questdb.mp.Sequence;
+import io.questdb.mp.SynchronizedJob;
+import io.questdb.std.*;
+import io.questdb.std.datetime.microtime.MicrosecondClock;
+import io.questdb.std.str.Path;
+import io.questdb.tasks.PartitionPurgeTask;
+
+import java.io.Closeable;
+import java.util.Comparator;
+import java.util.PriorityQueue;
+
+import static io.questdb.cairo.TableUtils.setPathForPartition;
+import static io.questdb.cairo.TableUtils.txnPartitionConditionally;
+
+public class PartitionPurgeJob extends SynchronizedJob implements Closeable {
```

Comment:
Why `O3PartitionPurgeJob` is not enough?

Reply:
It is indeed, the name mislead me, did not want to conflict and wanted to validate the strategy. I have removed the exceeding code (that I produced) and checked that the partition is purged.